src/llvm-alloc-helpers.cpp

```diff
@@ -125,6 +125,12 @@ JL_USED_FUNC void AllocUseInfo::dump(llvm::raw_ostream &OS)
     OS << "hastypeof: " << hastypeof << '\n';
     OS << "refload: " << refload << '\n';
     OS << "refstore: " << refstore << '\n';
+    OS << "allockind:";
+    if ((allockind & AllocFnKind::Uninitialized) != AllocFnKind::Unknown)
+      OS << " uninitialized";
+    if ((allockind & AllocFnKind::Zeroed) != AllocFnKind::Unknown)
```

Comment:
Do we ever set an allocation as `Zeroed?`

Reply:
We don't currently use this case. But could be relevant in the future hypothetically. This may be a bit awkward to have the function's behavior to depend upon a runtime value (I am not sure if these could be dropped or merged in the future). But that seemed a good approximation right now of what codegen wants to express (of whether the AllocFnKind is observable).